### PR TITLE
Added pre-commit job for neutron operator

### PR DESCRIPTION
**Pull request by amolkahat**
This patch adds pre-commit job for neutron
operator.

Signed-off-by: Amol Kahat <amolkahat@gmail.com>

### CODE REVIEW

File: ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml [hunk 1]
```diff
@@ -76,6 +81,19 @@ tests:
...
+  commands: |
+    export SKIP=no-commit-to-branch
+    export GOFLAG=
+    export PRE_COMMIT_HOME=/tmp
+    export GOLANGCI_LINT_CACHE=/tmp
+    mkdir -p ../operator/ && cp -r . ../operator
+    cd ../operator
+    pre-commit run -a --show-diff-on-failure
+  container:
+    clone: true
```
Comment (sandeepyadav93) [inline ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml:94]: @viroel do we need `clone: true` here , I don't see it in `openstack-operator`, it was removed with PR [2] so I want to cross-check if that was intentional.

https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/openstack-operator/openstack-k8s-operators-openstack-operator-master.yaml#L102-L103
[2] https://github.com/openshift/release/pull/36038/files

Comment (sandeepyadav93) [reply ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml:94]: Answering myself:

If we build the container from src we already get the cloned repo, but if we use a different container we need `clone:true` , As we are using `src`  to build` openstack-operator-ci-pre-commi`t we don't need `clone: true here.

Comment (amolkahat) [reply ci-operator/config/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master.yaml]: Removing `clone:true`

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-neutron-operator-master-precommit-check

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-neutron-operator-master-precommit-check

File: ci-operator/jobs/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -406,3 +406,51 @@ presubmits:
...
+    - ^master$
+    - ^master-
+    cluster: build02
+    context: ci/prow/precommit-check
+    decorate: true
+    labels:
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openstack-k8s-operators-neutron-operator-master-precommit-check
+    optional: true
```
Comment (marios) [inline ci-operator/jobs/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master-presubmits.yaml:421]: should precommit be optional? 

in my opinion it should not be but we should agree about this and be consistent across all the operator repos

**Review by marios — COMMENTED** [review]
OK but job is currently failing not sure on what though :)

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/36012/rehearse-36012-pull-ci-openstack-k8s-operators-neutron-operator-master-precommit-check/1637869405901688832

```

test -s /go/src/github.com/openstack-k8s-operators/operator/bin/golangci-lint || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.51.2
golangci/golangci-lint info checking GitHub for tag 'v1.51.2'
golangci/golangci-lint info found version: 1.51.2 for v1.51.2/linux/amd64
golangci/golangci-lint info installed ./bin/golangci-lint
/go/src/github.com/openstack-k8s-operators/operator/bin/golangci-lint run --fix
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make: *** [Makefile:117: golangci-lint] Error 4 
```

File: ci-operator/jobs/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master-presubmits.yaml (hunk 2 above)
Comment (amolkahat) [reply ci-operator/jobs/openstack-k8s-operators/neutron-operator/openstack-k8s-operators-neutron-operator-master-presubmits.yaml:421]: Ok. Marking it as voting. 

`golangci-lint` is failing, sometimes it passes and sometimes it fails.

**Comment by viroel** [comment]
Gibi is also working on adding this job[1], we can close this one.

[1] https://github.com/openshift/release/pull/38080

**Pull request closed without merging**